homeassistant/components/alexa/__init__.py

```diff
@@ -87,18 +87,12 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     config = config[DOMAIN]
 
-    flash_briefings_config = config.get(CONF_FLASH_BRIEFINGS)
-
     intent.async_setup(hass)
 
-    if flash_briefings_config:
+    if flash_briefings_config := config.get(CONF_FLASH_BRIEFINGS):
         flash_briefings.async_setup(hass, flash_briefings_config)
 
-    try:
-        smart_home_config = config[CONF_SMART_HOME]
-    except KeyError:
-        pass
-    else:
+    if smart_home_config := config.get(CONF_SMART_HOME):
```

Comment:
```suggestion
    # smart_home being absent is not the ame as smart_home being None
    if CONF_SMART_HOME in config:
        smart_home_config: dict[str, Any] | None = config[CONF_SMART_HOME]
```

Reply:
I suggest using the code above, with a comment explaining why we can't use a walrus here.
It seems that "key present and None" is not the same as "key absent".

Reply:
From [smart_home_setup_http.py](https://github.com/home-assistant/core/blob/dev/homeassistant/components/alexa/smart_home_http.py#L92):

> This is optional, triggered by having a `smart_home:` sub-section in the
    alexa configuration.
    Even if that's disabled, the functionality in this module may still be used
    by the cloud component which will call async_handle_message directly.